### PR TITLE
Split stdout and stderr in runCommand.  Bugs in pid1.

### PR DESCRIPTION
**Pull request by thockin**
A twofer.  In testing stdout/stderr I found bugs in pid1.

### CODE REVIEW

**Comment by thockin** [comment]
ping @stp-ip

**Comment by stp-ip** [comment]
Damn sorry got buried. Thanks for the ping. In my queue for today.

File: pkg/pid1/test/fast-exit/test.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+go build
+docker build -t example.com/pid1-fast-exit .
```
Comment (stp-ip) [inline pkg/pid1/test/fast-exit/test.sh]:
```suggestion
docker build -t example.com/fast-exit .
```

File: pkg/pid1/pid1.go [hunk 2]
```diff
@@ -25,38 +25,46 @@ func ReRun() error {
 	if err := cmd.Start(); err != nil {
 		return err
 	}
-	go runInit(cmd.Process.Pid)
-	return cmd.Wait()
+	runInit(cmd.Process.Pid)
+	return nil
```
[the remark this answers is not in the document]
Comment (thockin) [reply pkg/pid1/pid1.go]: Waiting here races with the delivery of SIGCHLD, which caused occasional panic() or errors returned to the caller.  We can't be trying to wait in two places, and SIGCHLD is coming either way, so this PR just moves it all to that path.

File: pkg/pid1/test/fast-exit/test.sh (hunk 1 above)
Comment (thockin) [reply pkg/pid1/test/fast-exit/test.sh]: ACK!  GOod catch

File: pkg/pid1/pid1.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (stp-ip) [reply pkg/pid1/pid1.go]: Makes sense. Thanks for the context.

**Pull request merged**